scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -122,12 +120,19 @@ def _focus_one_ws(
         mantid.DeleteWorkspace(solid_angle)
 
     # Focus the spectra into banks
-    focused_ws = mantid.DiffractionFocussing(InputWorkspace=aligned_ws, GroupingFileName=run_details.grouping_file_path)
+    focused_ws = mantid.DiffractionFocussing(
+        InputWorkspace=aligned_ws,
+        GroupingFileName=run_details.grouping_file_path,
+        OutputWorkspace=f"{run_number}_focused",
+    )
 
     instrument.apply_calibration_to_focused_data(focused_ws)
 
+    if instrument.get_instrument_prefix() == "OSIRIS":
+        focused_ws = instrument.apply_drange_cropping(run_number, focused_ws)
```

Comment:
Please can you define a `apply_drange_cropping` function in abstract_inst.py that doesn't do anything. The definition in osiris.py will then be an override and you will be able to remove the if instrument="OSIRIS" statement on L131

Reply:
Updated